arch-linux/archlinux_test.go

```diff
@@ -0,0 +1,77 @@
+package arch_linux
+
+import (
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUpdate(t *testing.T) {
+	testCases := []struct {
+		name                   string
+		inputJSONFile          string
+		expectedOutputJSONFile string
+		expectedError          string
+		ArchLinuxServerUrl     string
+	}{
+		{
+			name:                   "happy path",
+			inputJSONFile:          "testdata/archlinux.json",
+			expectedOutputJSONFile: "testdata/AVG-4.json",
+		},
+		{
+			name:               "sad path, unreachable Arch Linux service",
+			expectedError:      "failed to retrieve Arch Linux CVE details",
+			ArchLinuxServerUrl: "http://foo/bar/baz",
+		},
+		{
+			name:          "sad path, invalid json",
+			inputJSONFile: "testdata/invalid.json",
+			expectedError: "failed to retrieve Arch Linux CVE details",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			var ArchLinuxURL string
+			if tc.ArchLinuxServerUrl != "" {
+				ArchLinuxURL = tc.ArchLinuxServerUrl
+			} else {
+				ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					b, _ := ioutil.ReadFile(tc.inputJSONFile)
+					_, _ = io.WriteString(w, string(b))
+				}))
+				ArchLinuxURL = ts.URL
+				defer func() {
+					ts.Close()
+				}()
+			}
+
+			dir, _ := ioutil.TempDir("", "TestUpdate-*")
+			defer func() {
+				_ = os.RemoveAll(dir)
+			}()
```

Comment:
```suggestion
			dir := t.TempDir()
```

https://golang.org/pkg/testing/#T.TempDir